src/rgw/driver/rados/rgw_notify.cc

```diff
@@ -960,13 +962,25 @@ static inline bool notification_match(reservation_t& res,
 		      reservation_t& res,
 		      const RGWObjTags* req_tags)
 {
-  const RGWPubSub ps(res.store, res.user_tenant);
-  const RGWPubSub::Bucket ps_bucket(ps, res.bucket);
   rgw_pubsub_bucket_topics bucket_topics;
-  auto rc = ps_bucket.get_topics(res.dpp, bucket_topics, res.yield);
-  if (rc < 0) {
-    // failed to fetch bucket topics
-    return rc;
+  if (res.store->svc()->zone->get_zonegroup().supports(
+          rgw::zone_features::notification_v2)) {
+    rgw::sal::Attrs attrs = res.bucket->get_attrs();
```

Comment:
```suggestion
    const rgw::sal::Attrs& attrs = res.bucket->get_attrs();
```

Reply:
i replied this earlier to Casey, adding const is resulting in compilation error, the iter.find() calls fails in compilation with const attrs. i have change the code to use reference instead of copying.

Reply:
try adding `const` to `bufferlist& bl = iter->second;` below

Reply:
i had put auto, if i recollect, lemme retry

Reply:
> try adding `const` to `bufferlist& bl = iter->second;` below

thanks that worked, i think i did not add const auto& everywhere and that  was the cause for compilation error. Done